packages/insomnia-app/app/models/request.js

```diff
@@ -294,8 +282,9 @@ export function updateMimeType(
   // 1. Update Content-Type header //
   // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ //
 
-  const hasBody = typeof mimeType === 'string';
-  if (!hasBody || mimeType === CONTENT_TYPE_OTHER) {
+  if (mimeType === CONTENT_TYPE_NO_BODY) {
```

Comment:
You pretty much got this right, although it's a bit simpler than what you did. Technically the `Request.body` attribute can be `{}`. This is why the `hasBody` variable was set to `typeof mimeType === 'string'`. 

That means that your change can be simplified to the following:

```suggestion
  if (!hasBody) {
    headers = headers.filter(h => h !== contentTypeHeader);
  if (mimeType === CONTENT_TYPE_OTHER) {
```

Reply:
I think this is actually the only change needed for this PR besides updating the tests.